docs/hugo/content/design/ADR-2023-02-Helm-Chart-Size-Limitations.md

```diff
@@ -0,0 +1,108 @@
+---
+title: '2023-01: Helm Chart Size Limitations and Workarounds'
+---
+
+## Context
+
+Helm v3 stores chart state in a Kubernetes secret by default. Kubernetes secrets are limited to a max size based on 
+etcd's configuration. The default etcd configuration limits secrets to a maximum size of 1MB.
+
+Our chart secret is fast approaching 1MB. In fact, 
+[outstanding PRs](https://github.com/Azure/azure-service-operator/pull/2727) already cause us to surpass the 1MB limit.
+
+Once this limit is reached, users will be unable to install our chart. When this happens, the error looks like:
+> Error: create: failed to create: Secret "sh.helm.release.v1.asov2.v1" is invalid: data: Too long: must have at most 1048576 bytes
+
+Digging in a bit, the Helm secret contains compressed data. This means that our chart itself can consume >1MB as long 
+as its compressed size is <1MB. Here's a breakdown of our chart size right now:
+
+| Scenario                         | Chart size | Helm secret size | Compression ratio |
+|----------------------------------|------------|------------------|-------------------|
+| v2.0.0-beta.4                    | 9917 KB    | 732 KB           | 0.07              |
+| v2.0.0-beta.4 (all docs removed) | 8600 KB    | 536 KB           | 0.06              |
```

Comment:
At first glance, I thought this table was indicating compression didn't achieve very much. I usually see _Compression ratio_ as _how much was saved_.

```suggestion
| Scenario                         | Chart size | Helm secret size | Compressed Size | Compression|
|----------------------------------|-----------:|-----------------:|----------------:|-----------:|
| v2.0.0-beta.4                    | 9917 KB    | 732 KB           | 7.3%            |13.5x       |
| v2.0.0-beta.4 (all docs removed) | 8600 KB    | 536 KB           | 6.2%            |16.0x       |
```

Reply:
Ah, I was using the [compression ratio](https://en.wikipedia.org/wiki/Data_compression_ratio) - I agree you don't see it super often though I can rephrase. I might convert to the Space Saving there (which is just 1- this value)